CMakeLists.txt

```diff
@@ -22,7 +40,16 @@ endif (WIN32)
 set(ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 
 # Where the version source file for xplat is generated
-set(VERSION_FILE_PATH "${CMAKE_BINARY_DIR}/version.cpp")
+set(VERSION_FILE_PATH "${CMAKE_BINARY_DIR}/version.c")
+
+if (NOT WIN32)
+  # Drop the static scope for SOS, so it's available outside the
+  # compilation unit for external linkage; see extern declaration
+  # in SOS sources.
+  file(READ "${VERSION_FILE_PATH}" VERSION_FILE_CONTENTS)
+  string(REPLACE "static char" "char" VERSION_LINE_WITHOUT_STATIC "${VERSION_FILE_CONTENTS}")
+  file(WRITE "${VERSION_FILE_PATH}" "${VERSION_LINE_WITHOUT_STATIC}")
```

Comment:
@jkoritzinsky, in diagnostics repo, we need to define `char sccsid[]` as non-static, so it is visible outside the compilation unit (file) and satisfies this extern: https://github.com/dotnet/diagnostics/blob/main/src/SOS/Strike/strike.cpp#L10899

since I haven't changed anything in eng/native, I worked around it from call-site. Perhaps we can add its support in version generator whether or not to omit `static, or just omit `static` everywhere since we have `__attribute__((used))` in version.c, so it remains in binary after stripping (e.g. `strings libsos.so | grep '@(#)'` shows the version even with non-static).